src/app/core/services/config/protocol.service.ts

```diff
@@ -1,21 +1,33 @@
+// tslint:disable: forin
 import { HttpClient } from '@angular/common/http'
 import { Injectable } from '@angular/core'
 
 import {
+  DefaultParticipantAttributeOrder,
   DefaultProtocolBranch,
   DefaultProtocolEndPoint,
+  DefaultProtocolGithubRepo,
   DefaultProtocolPath,
   DefaultQuestionnaireFormatURI,
-  DefaultQuestionnaireTypeURI
+  DefaultQuestionnaireTypeURI,
+  GIT_API_URI
 } from '../../../../assets/data/defaultConfig'
 import { ConfigKeys } from '../../../shared/enums/config'
 import { Assessment } from '../../../shared/models/assessment'
+import { sortObject } from '../../../shared/utilities/sort-object'
 import { LogService } from '../misc/log.service'
 import { RemoteConfigService } from './remote-config.service'
 import { SubjectConfigService } from './subject-config.service'
 
 @Injectable()
 export class ProtocolService {
+  GIT_TREE = 'tree'
+  GIT_BRANCHES = 'branches'
+  ATTRIBUTE_KEY = 0
+  ATTRIBUTE_VAL = 1
+  // NOTE: If no/null attribute order, default value:
+  HIGH_ATTRIBUTE_ORDER = 99
```

Comment:
perhaps use something higher. you can even use the max value of an int `Number.MAX_SAFE_INTEGER` which is the value of int 2^53 - 1 (because the memory is already allocated for an int). Otherwise, use something usually high.